File: backend/src/Core/Models/OptionsChain.cs
namespace Core.Models;

public class OptionsChain
{
    public required FullOptionsChain Options { get; set; }
}

public class FullOptionsChain
{
    public required List<StrikePriceData> Option { get; set; }
}

public class StrikePriceData
{
    public required string Symbol { get; set; }
    public string? Description { get; set; }
    public string? Underlying { get; set; }
    public float Strike { get; set; }
    public int OpenInterest { get; set; }
    public string? ExpirationDate { get; set; }
    public string? OptionType { get; set; }
}

// public class Greeks
// {
//     public double Delta { get; set; }
// }

File: apps/web/src/app/shared/models/options/options-chain.ts
export interface OptionsChain {
  optionsChain: StrikePriceData[];
}

export interface StrikePriceData {
  symbol: string;
  description?: string;
  underlying?: string;
  strike: number;
  volume: number;
  openInterest: number;
  expirationDate?: string;
  optionType?: string;
  last?: number;
  bid?: number;
  ask?: number;
  greeks?: Greeks;
  impliedVolatility?: ImpliedVolatility;
  activity?: Activity;
}

export interface Greeks {
  delta: number;
  gamma: number;
  theta: number;
  vega: number;
  rho: number;
  phi: number;
  updatedAt: string;
}

export interface ImpliedVolatility {
  bidIv: number;
  midIv: number;
  askIv: number;
  smvVol: number;
  updatedAt: string;
}

export interface Activity {
  openInterest: number;
  volume: number;
  unusualActivity: number;
}
